src/Orleans/CodeGeneration/GrainFactoryBase.cs

```diff
@@ -168,9 +183,9 @@ public static class GrainFactoryBase
             {
                 throw new ArgumentException("Cannot fabricate grain-reference for non-grain type: " + grainType.FullName);
             }
-            int grainTypeCode = TypeCodeMapper.GetImplementationTypeCode(interfaceId, grainClassNamePrefix);
+            int grainTypeCode = TypeCodeMapper.GetImplementationTypeCode(grainType, grainClassNamePrefix);
```

Comment:
This one word is the key of this change: use the Type (this is interface type) instead of the interface id code.